### PR TITLE
Fixes pathValueMerge to merge and follow references.

### PR DESCRIPTION
**Pull request by trxcllnt**

### CODE REVIEW

**Comment by jhusain** [comment]
Reviewed this and the change looks good. @michaelbpaulson waiting on signoff from you.

**Comment by michaelbpaulson** [comment]
@trxcllnt @jhusain the path value merge was intentionally never suppose to follow references.  All path values should be reported without references.

Notice below, that no references should ever be encountered in a path.
```javascript
var route = new Router({
    route: 'lolomo[{integer:indices}]',
    get: function(alias) {
        return alias.indices.reduce(function(acc, curr) {
            acc.push({
                path: ['lolomo', curr],
                value: $ref(toSomeList())
            });
            return acc;
        }, []);
    }
});
```

Even with higher precedence paths, the same holds true, the JSON-Graph/PathValue return never consider references, just flat merge.

**Comment by trxcllnt** [comment]
@jhusain can we merge this commit for now, and decide what to do about it later?

File: src/cache/pathValueMerge.js [hunk 1]
```diff
@@ -44,70 +44,49 @@ module.exports = function pathValueMerge(cache, pathValue) {
...
-    do {
+            curr = prev[key];
 
-        cloned = clone(pathValue.value);
-        curr[key] = cloned;
+            if (branch) {
+                if (!curr) {
+                    curr = prev[key] = {};
+                }
+                curr = innerPathValueMerge(root, curr, value, path, depth + 1, reference);
```
Comment (michaelbpaulson) [inline src/cache/pathValueMerge.js]: Exceeds 80 characters, please ensure eslint passes (of which, apparently, the build is not checking for).

File: src/cache/pathValueMerge.js [hunk 2]
```diff
@@ -44,70 +44,49 @@ module.exports = function pathValueMerge(cache, pathValue) {
...
+            curr = innerPathValueMerge(root, root, value, curr.value, 0, true);
+            type = curr.$type;
         }
-    }
-
 
-    // TODO: This clearly needs a re-write.  I am just unsure of how i want
-    // this to look.  Plus i want to measure performance.
-    outerKey = path[i];
+        if (type === void 0) {
```
Comment (michaelbpaulson) [inline src/cache/pathValueMerge.js]: no `void 0` do `undefined` for explicitness sake considering void actually evaluates the `0` as an expression, executes it, and ignores the output, but instead returns undefined.

**Comment by angelf** [comment]
> As part of the roadmap we are exploring ways of adding better support to
Falcor for  heterogeneous lists.

I am very interested in this scenario. Was the roadmap already published?

**Pull request closed without merging**